src/xtgeo/grid3d/_grid3d_utils.py

```diff
@@ -77,52 +78,45 @@ def scan_dates(pfile, maxdates=1000, dataframe=False):
     return zdates
 
 
-def _scan_ecl_keywords(pfile, maxkeys=100000, dataframe=False):
-
-    ultramax = int(1000000 / 9)  # cf *swig_bnd_char_1m in cxtgeo.i
-    if maxkeys > ultramax:
-        raise ValueError("maxkeys value is too large, must be < {}".format(ultramax))
-
-    rectypes = _cxtgeo.new_intarray(maxkeys)
-    reclens = _cxtgeo.new_longarray(maxkeys)
-    recstarts = _cxtgeo.new_longarray(maxkeys)
-
+def _scan_ecl_keywords(pfile, maxkeys=MAXKEYWORDS, dataframe=False):
+    logger.info("Scanning ECL keywords...")
     cfhandle = pfile.get_cfhandle()
 
-    nkeys, keywords = _cxtgeo.grd3d_scan_eclbinary(
-        cfhandle, rectypes, reclens, recstarts, maxkeys
+    # maxkeys*10 is used for 1D kewords; 10 => max 8 letters in eclipse +
+    # "|" + "extra buffer"
+    nkeys, keywords, rectypes, reclens, recstarts = _cxtgeo.grd3d_scan_eclbinary(
+        cfhandle, maxkeys * 10, maxkeys, maxkeys, maxkeys
     )
 
-    if nkeys < 0:
-        raise ValueError(f"scanning ecl keywords exited with error code {nkeys}")
-
     pfile.cfclose()
 
+    if nkeys == -1:
+        raise XTGeoCLibError(f"scanning ecl keywords exited with error code {nkeys}")
+
+    if nkeys == -2:
+        raise RuntimeError(
```

Comment:
Is this really a RuntimeError? Seems more like a ValueError to me?

Reply:
Since the user currently cannot control the number of MAXKEYWORDS, I was thinking of RuntimeError. Perhaps chaning it to XTGeoCLIBError? I'll do that